tests/fine_tuning/test_lora.py

```diff
@@ -0,0 +1,112 @@
+import os
+from pathlib import Path
+
+import pytest
+from torch import nn
+
+from modalities.config.config import load_app_config_dict
+from modalities.models.lora.lora_layers import LoRALinear
+from modalities.models.lora.utils import convert_layer, conversion_lora
+from modalities.models.model import NNModel
+from modalities.models.utils import ModelTypeEnum, get_model_from_config
+from tests.conftest import _ROOT_DIR
+
+
+@pytest.fixture()
+def set_env():
+    os.environ["LOCAL_RANK"] = "0"
+    os.environ["RANK"] = "0"
+    os.environ["WORLD_SIZE"] = "1"
+
+
+@pytest.fixture()
+def config_file_path() -> Path:
+    config_file_path = _ROOT_DIR / Path("tests/fine_tuning/test_configs/" + "config_lorem_ipsum_sft.yaml")
+    return config_file_path
+
+
+@pytest.fixture()
+def config_dict(config_file_path: Path) -> dict:
+    return load_app_config_dict(config_file_path=config_file_path)
+
+
+@pytest.fixture()
+def model(set_env, config_dict: dict) -> NNModel:
+    return get_model_from_config(config=config_dict, model_type=ModelTypeEnum.MODEL)
+
+
+@pytest.fixture()
+def r():
+    return 8
+
+
+@pytest.fixture()
+def alpha():
+    return 1
+
+
+@pytest.fixture()
+def layer_types():
+    return ["attn"]
```

Comment:
maybe here also add the other one and make the tests run for both / parametrize it...